spec/javascripts/accordion-with-descriptions-spec.js

```diff
@@ -0,0 +1,181 @@
+            </div>\
+            <div class="subsection__content" id="subsection_content_0">\
+              <ul class="subsection__list">\
+                <li class="subsection__list-item">\
+                  <a href="">Subsection list item in here</a>\
+                </li>\
+              </ul>\
+            </div>\
+          </div>\
+          <div class="subsection">\
+            <div class="subsection__header">\
+              <h2 class="subsection__title">Subsection title in here</h2>\
+              <p class="subsection__description">Subsection description in here</p>\
+            </div>\
+            <div class="subsection__content" id="subsection_content_1">\
+              <ul class="subsection__list">\
+                <li class="subsection__list-item">\
+                  <a href="">Subsection list item in here</a>\
+                </li>\
+              </ul>\
+            </div>\
+          </div>\
+        </div>\
+      </div>');
+
+    var accordion = new GOVUK.Modules.AccordionWithDescriptions();
+    accordion.start($element);
+  });
+
+  afterEach(function() {
+    $(document).off();
+  });
+
+  // Setup
+  // Add & remove classes to show the JS has worked
+  // Add an 'Open all' button
+  // Set the correct text and aria attributes (aria-expanded, aria-controls) for the button
+
+  // Add a class .js-accordion-with-descriptions
+  it("has a class of js-accordion-with-descriptions", function () {
+    expect($element).toHaveClass("js-accordion-with-descriptions");
+  });
+
+  // Remove the class .js-hidden
+  it("does not have a class of js-hidden", function () {
+    expect($element).not.toHaveClass("js-hidden");
+  });
+
+  // Add a subsection controls div, with a class of .js-subsection controls
+  it("has a child element with a class of subsection-controls", function () {
+    expect($element).toContain('.js-subsection-controls');
+  });
+
+  // Insert a button inside .js-subsection-controls
+  it("has a child element which is a button", function () {
+    expect($element).toContain('.js-subsection-controls button');
+  });
+
+  // Set the aria-expanded attribute to false, as all subsections are initially closed
+  it("has an open/close all button with an aria-expanded attribute and it is false", function () {
+    var $button = $element.find('.js-subsection-controls button');
+
+    expect($button).toHaveAttr("aria-expanded", "false");
+  });
+
+  // Set the text inside the button to "Open all"
+  it("has an open/close all button with text inside which is equal to Open all", function () {
+    var $openCloseAllButton = $element.find('.js-subsection-controls button');
+
+    expect($openCloseAllButton).toHaveText("Open all");
+  });
+
+  // Ensure that the value for the aria-controls attribute matches the IDs of all of the subsections
+  it("has a value for the aria-controls attribute which includes all of the subsection_content_IDs", function () {
```

Comment:
I find the comment doesn't add much value on top of the example description here. I'd just bin the comment?